dvc/utils/http.py

```diff
@@ -12,9 +16,19 @@ def open_url(url, mode="r", encoding=None):
     Url could be a string or a callable returning a string.
     """
     assert mode in {"r", "rt", "rb"}
+    context = iter_url(url)
+    with context as (response, it):
+
+        def retry():
+            (
+                _,
+                iterator,
+            ) = context.gen.throw(  # pylint: disable=maybe-no-member
+                Reiterate()
+            )
+            return iterator
 
-    with iter_url(url) as (response, it):
-        bytes_stream = IterStream(it)
+        bytes_stream = IterStream(it, retry=retry)
```

Comment:
This creates an impression of using retries, which it isn't.

Reply:
fixed to recreate_iterator